include/gsl/gsl_byte

```diff
@@ -42,15 +44,15 @@
 
 #ifndef GSL_USE_STD_BYTE
 // this tests if we are under MSVC and the standard lib has std::byte and it is enabled
-#if defined(_HAS_STD_BYTE) && _HAS_STD_BYTE
+#if (defined(_HAS_STD_BYTE) && _HAS_STD_BYTE) || (defined(__cpp_lib_byte) && __cpp_lib_byte >= 201603)
 
 #define GSL_USE_STD_BYTE 1
 
-#else // defined(_HAS_STD_BYTE) && _HAS_STD_BYTE
+#else // defined(_HAS_STD_BYTE) && _HAS_STD_BYTE || (defined(__cpp_lib_byte) && __cpp_lib_byte)
```

Comment:
The `#else` and `#endif` comments say `&& __cpp_lib_byte)` instead of `&& __cpp_lib_byte >= 201603)`.

Reply:
I'll make that more consistent and get another version pushed out.

Reply:
Updated